src/librustdoc/html/markdown.rs

```diff
@@ -931,15 +930,17 @@ impl Markdown<'_> {
         if md.is_empty() {
             return String::new();
         }
-        let replacer = |_: &str, s: &str| {
-            if let Some(link) = links.iter().find(|link| &*link.original_text == s) {
-                Some((link.href.clone(), link.new_text.clone()))
+        let mut replacer = |broken_link: BrokenLink<'_>| {
+            if let Some(link) =
+                links.iter().find(|link| &*link.original_text == broken_link.reference)
+            {
+                Some((CowStr::Borrowed(&link.href), CowStr::Borrowed(&link.new_text)))
```

Comment:
I don't know if you did this to make it explicit, but it might be better to use `Into` (it's what marcusklaas suggested to me):

```suggestion
                Some((link.href.as_str().into(), link.new_text.as_str().into()))
```

Reply:
See here: https://github.com/rust-lang/rust/issues/76687#issuecomment-691738332